include/zephyr/audio/midi.h

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) 2024 Titouan Christophe
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_USB_CLASS_MIDI_H_
+#define ZEPHYR_INCLUDE_USB_CLASS_MIDI_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+
+/**
+ * @brief Universal MIDI Packet definitions
+ * @defgroup usb_midi_ump USB MIDI 2.0 Universal MIDI Packet definitions
+ * @ingroup usb
+ * @since 4.1
+ * @version 0.1.0
+ * @see ump112: "Universal MIDI Packet (UMP) Format and MIDI 2.0 Protocol"
+ *              Document version 1.1.2
+ * @{
+ */
+
+/**
+ * @brief      Universal MIDI Packet container
+ */
+struct midi_ump {
+	uint32_t data[4]; /**< Raw content */
+};
```

Comment:
@titouanc Not sure how other interfaces would require it, e.g. Bluetooth. You could change usbd_midi_send() to int `usbd_midi_send(const struct device *dev, const struct midi_ump ump);`, so it would be more safe, and then I would also be happy.

Also, please add a comment that the variables must be in the CPU native endianness.